app/src/main/java/org/wikipedia/analytics/eventplatform/EventPlatformClient.java

```diff
@@ -0,0 +1,451 @@
+package org.wikipedia.analytics.eventplatform;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Timer;
+import java.util.TimerTask;
+
+import static org.wikipedia.analytics.eventplatform.EventPlatformClientIntegration.deleteStoredSessionId;
+import static org.wikipedia.analytics.eventplatform.EventPlatformClientIntegration.fetchStreamConfigs;
+import static org.wikipedia.analytics.eventplatform.EventPlatformClientIntegration.getAppInstallId;
+import static org.wikipedia.analytics.eventplatform.EventPlatformClientIntegration.getIso8601Timestamp;
+import static org.wikipedia.analytics.eventplatform.EventPlatformClientIntegration.getStoredSessionId;
+import static org.wikipedia.analytics.eventplatform.EventPlatformClientIntegration.getStoredStreamConfigs;
+import static org.wikipedia.analytics.eventplatform.EventPlatformClientIntegration.isOnline;
+import static org.wikipedia.analytics.eventplatform.EventPlatformClientIntegration.postEvents;
+import static org.wikipedia.analytics.eventplatform.EventPlatformClientIntegration.setStoredSessionId;
+import static org.wikipedia.analytics.eventplatform.EventPlatformClientIntegration.setStoredStreamConfigs;
+import static org.wikipedia.analytics.eventplatform.SamplingConfig.Identifier.DEVICE;
+import static org.wikipedia.analytics.eventplatform.SamplingConfig.Identifier.PAGEVIEW;
+import static org.wikipedia.analytics.eventplatform.SamplingConfig.Identifier.SESSION;
+import static org.wikipedia.settings.Prefs.isEventLoggingEnabled;
+
+public final class EventPlatformClient {
+
+    /**
+     * Stream configs to be fetched on startup and stored for the duration of the application
+     * lifecycle.
+     */
+    static Map<String, StreamConfig> STREAM_CONFIGS = new HashMap<>();
+
+    /*
+     * When ENABLED is false, items can be enqueued but not dequeued.
+     * Timers will not be set for enqueued items.
+     * QUEUE will not grow beyond MAX_QUEUE_SIZE.
+     *
+     * Inputs: network connection state on/off, connection state bad y/n?
+     * Taken out of iOS client, but flag can be set on the request object to wait until connected to send
+     */
+    private static boolean ENABLED = isOnline() && isEventLoggingEnabled();
```

Comment:
@sharvaniharan  Hint 2: `isEventLoggingEnabled()` accesses the app's Preferences, which are not mockable, and will cause issues with tests.  More generally the check for `isEventLoggingEnabled()` should probably not be done here anyway, but right at the point where the data is about to be sent.